include/swift/AST/DiagnosticsSema.def

```diff
@@ -1281,6 +1281,16 @@ WARNING(expr_dynamic_lookup_swift3_objc_inference,none,
         "deprecated in Swift 4",
         (DescriptiveDeclKind, DeclName, Identifier))
 
+ERROR(inherited_default_value_not_on_constructor_param,none,
+      "default value inheritance via 'super' is only valid for initializer "
+      "parameters", ())
+ERROR(corresponding_param_not_defaulted,none,
+      "default value inheritance via 'super' requires that the corresponding "
+      "parameter of the overridden initializer has a default value", ())
+ERROR(inherited_default_value_used_in_non_overriding_constructor,none,
+      "default value inheritance via 'super' can only be used on the "
+      "parameters of overriding initializers", ())
```

Comment:
What happens for inherited convenience initializers (which are only sort of overriding)?

Reply:
Inherited convenience inits are visible via name lookup like other inherited decls when the right flag on the ClassDecl is set. Only designated inits get explicitly synthesized overrides.

Reply:
I've updated the checks and wording of the diagnostics to restrict this to designated inits now.